packages/js-sdk/src/sandbox/filesystem/index.ts

```diff
@@ -141,40 +125,44 @@ export class Filesystem {
     return res.data
   }
 
-  /**
-   * Writes content to a file on the path.
-   *   When writing to a file that doesn't exist, the file will get created.
-   *   When writing to a file that already exists, the file will get overwritten.
-   *   When writing to a file that's in a directory that doesn't exist, you'll get an error.
-   *
-   * @param path Path to a new file. For example '/dirA/dirB/newFile.txt' when creating 'newFile.txt'
-   * @param data Data to write to a new file
-   * @param opts Options for the request
-   * @returns Information about the written file
-   */
+  async write(path: string, data: WriteData, opts?: FilesystemRequestOpts): Promise<EntryInfo>
+  async write(files: { path: string; data: WriteData }[], opts?: FilesystemRequestOpts): Promise<EntryInfo[]>
   async write(
-    path: string,
-    data: string | ArrayBuffer | Blob | ReadableStream,
+    pathOrFiles: string | { path: string; data: WriteData }[],
+    dataOrOpts?: WriteData | FilesystemRequestOpts,
     opts?: FilesystemRequestOpts
-  ): Promise<EntryInfo> {
-    const blob = await new Response(data).blob()
+  ): Promise<EntryInfo | EntryInfo[]> {
+    if (typeof pathOrFiles === 'string' && Array.isArray(dataOrOpts)) {
+      throw new Error('Cannot specify path with array of files')
+    }
+
+    const { path, writeOpts, writeFiles } =
+      typeof pathOrFiles === 'string'
+        ? { path: pathOrFiles, writeOpts: opts, writeFiles: [{ data: dataOrOpts }] }
+        : { path: undefined, writeOpts: dataOrOpts, writeFiles: pathOrFiles }
+
+    const blobs = await Promise.all(
+      (writeFiles as { path: string; data: WriteData }[]).map((f) => new Response(f.data).blob())
```

Comment:
It might not be possible but I'm still thinking is we need to use `as` anywhere — could the typeguarding get rid of the need so we can be sure the type is correct without `as`suming. This is not blocking, just would make it extra nice.